app/[[...slug]]/page.tsx

```diff
@@ -4,20 +4,29 @@ import React from "react"
 import loadPage from "@/app/[[...slug]]/loadPage"
 import BasicPageLayout from "@/components/pages/basicPageLayout/BasicPageLayout"
 import goConfig from "@/lib/config/goConfig"
-import { GetPageByPathQuery, NodeGoPage } from "@/lib/graphql/generated/dpl-cms/graphql"
+import { NodeGoPage } from "@/lib/graphql/generated/dpl-cms/graphql"
+import { setPageMetadata } from "@/lib/helpers/helper.metadata"
 
-async function page(props: { params: Promise<{ slug: string[] }> }) {
-  const params = await props.params
-  const { slug } = params
-
-  let data: GetPageByPathQuery
-  // If no slug is provided, load the frontpage
+async function getPage(slug: string[]) {
   if (!slug) {
-    data = await loadPage(goConfig("routes.frontpage"))
-  } else {
-    const slugString = slug.join("/")
-    data = await loadPage(slugString)
+    return await loadPage(goConfig("routes.frontpage"))
+  }
+  const slugString = slug.join("/")
+  return await loadPage(slugString)
+}
+
+export async function generateMetadata(props: { params: Promise<{ slug: string[] }> }) {
```

Comment:
Consider moving this to a helper somewhere, as it would probably be reused somewhere on articles or other types of pages (?)